### PR TITLE
Support HTTP Basic Authentication

### PR DESCRIPTION
**Pull request by neekolas**
Adds support for 'login' event.
See: https://github.com/atom/electron/pull/3250

Adds a new option defaultAuthentication

### CODE REVIEW

File: Readme.md [hunk 1]
```diff
@@ -144,6 +144,15 @@ var nightmare = Nightmare({
 });
 ```
 
+##### defaultAuth
+A username and password pair for HTTP basic auth on all requests
+
+```js
+var nightmare = Nightmare({
+  defaultAuth: 'username:password'
```
Comment (justinmchase) [inline Readme.md]: I feel like this might be better as a more nested object like:
```
auth: {
  basic: {
    user: '...',
    pass: '...'
  }
}
```

**Comment by rosshinkley** [comment]
I think this is a good inclusion, provided it's unit tested.  I think using something like [`basic-auth`](https://www.npmjs.com/package/basic-auth) with the test Express server would work.

**Comment by brrd** [comment]
Looks like a duplicate #475

**Comment by justinmchase** [comment]
#475 looks like a simpler version of this with tests already.

**Comment by rosshinkley** [comment]
@brrd I'm sorry, I must have missed #475.

@justinmchase Agreed, closing as duplicate.

**Pull request closed without merging**